de.dashup.shared/src/main/java/de/dashup/shared/Widget.java

```diff
@@ -150,13 +157,31 @@ public void setCode(String code, Size size) {
     }
 
     public String getCodeWithWrapper() {
-        return "<div class=\"col " + this.size.getStyleClass() + "\">" +
-                "<div class=\"widget card\">" +
-                "<div class=\"card-content\">" +
-                this.getCode() +
-                "</div>" +
-                "</div>" +
-                "</div>";
+        StringBuilder html = new StringBuilder()
+                .append("<div class=\"card col ")
+                .append(this.size.getStyleClass())
+                .append("\">")
+                .append("<div class=\"card-content\">");
+        if (this.getProperties().size() > 0) {
+            Document document = Jsoup.parse(this.getCode());
+            for (Map.Entry<String, Property> propertyEntry : this.getProperties().entrySet()) {
```

Comment:
Can you please explain the following 5 lines to me? I think you are trying to insert the properties, but what on earth does the ternary operator check for / do?